mobile/src/viewModels/HomeViewModel.ts

```diff
@@ -168,21 +178,57 @@ export default class HomeViewModel {
         ];
     }
 
+    private submitPendingShortIfTimeForFull() {
+        if (AppController.Instance.User.localSettings?.current?.qol?.pendingShortQol && this.isTimeForFullQol()) {
+            this._settings.updateQolSettings({ pendingShortQol: false }, 'pendingShortQol');
+        }
+    }
+
     // returns true if it has been 28 calendar days since last Full QoL
+    // return true if there is a pending Full QoL
     private isTimeForFullQol(): boolean {
         const lastFullQol: Date = new Date(AppController.Instance.User.localSettings?.current?.qol?.lastFullQol);
         let nextFullQol: Date = lastFullQol;
         nextFullQol.setDate(nextFullQol.getDate() + 28);
         const today: Date = new Date();
+
         if (nextFullQol.getDay() === today.getDay() && nextFullQol.getMonth() === today.getMonth()
         && nextFullQol.getFullYear() === today.getFullYear()) {
-            this._settings.updateLastFullQol({ lastFullQol: Date() });
-            this._settings.updatePendingFullQol({ pendingFullQol: true });
+            this._settings.updateQolSettings({ lastFullQol: Date() }, 'lastFullQol');
+            this._settings.updateQolSettings({ pendingFullQol: true }, 'pendingFullQol');
             return true;
-        } else if (AppController.Instance.User.localSettings?.current?.qol?.pendingFullQol) { return true; }
+        } else if (AppController.Instance.User.localSettings?.current?.qol?.pendingFullQol) {
+            return true; 
+        }
         return false;
     }
 
+        // returns true if it has been 7 calendar days since last Short QoL
+        // return true if there is a pending Short QoL
+        private isTimeForShortQol(): boolean {
+            const lastShortQol: Date = new Date(AppController.Instance.User.localSettings?.current?.qol?.lastShortQol);
+            let nextShortQol: Date = lastShortQol;
+            nextShortQol.setDate(nextShortQol.getDate() + 7);
+            const today: Date = new Date();
+
+            if (nextShortQol.getDay() === today.getDay() && nextShortQol.getMonth() === today.getMonth()
+            && nextShortQol.getFullYear() === today.getFullYear()) {
+                this._settings.updateQolSettings({ lastShortQol: Date() }, 'lastShortQol');
+                this._settings.updateQolSettings({ pendingShortQol: true }, 'pendingShortQol');
+                return true;
+            } else if (AppController.Instance.User.localSettings?.current?.qol?.pendingShortQol) { 
+                return true; 
+            }
+            return false;
```

Comment:
This probably returns a boolean on it's own, I'm not sure if it would be too long, but you could probably do a `return (case 1) || (case 2)` here.